### DISCUSSION TITLE
bind exception in Zookeeper , when strimzi operator is upgraded from 0.31.1 to 0.36.1

### DISCUSSION THREAD

**Discussion by Janardhan78** [strimzi/strimzi-kafka-operator#9552] [Q&A]
We are facing an issue when we upgrade the strimzi operator from 0.31.1 to 0.36.1 , the zookeeper pods try to bind to old ip which was present before  the upgrade and due to this after operator upgrade it fails. Is this an known issue?

{"version": "1.2.0", "timestamp": "2024-01-12T09:27:16.976+00:00", "severity": "error", "service_id": "cluster-a", "message": "Couldn't bind to no-metrics-pattern-zookeeper-0.no-metrics-pattern-zookeeper-nodes.con.svc/192.168.147.217:2888 java.net.BindException: Cannot assign requested address|\tat java.base/sun.nio.ch.Net.bind0(Native Method)|\tat java.base/sun.nio.ch.Net.bind(Net.java:572)|\tat java.base/sun.nio.ch.Net.bind(Net.java:561)|\tat java.base/sun.nio.ch.NioSocketImpl.bind(NioSocketImpl.java:648)|\tat java.base/java.net.ServerSocket.bind(ServerSocket.java:388)|\tat java.base/java.net.ServerSocket.bind(ServerSocket.java:342)|\tat org.apache.zookeeper.server.quorum.Leader.createServerSocket(Leader.java:315)|\tat org.apache.zookeeper.server.quorum.Leader.lambda$new$0(Leader.java:294)|\tat java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)|\tat java.base/java.util.concurrent.ConcurrentHashMap$KeySpliterator.forEachRemaining(ConcurrentHashMap.java:3573)|\tat java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)|\tat java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)|\tat java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)|\tat java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)|\tat java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)|\tat java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)|\tat org.apache.zookeeper.server.quorum.Leader.<init>(Leader.java:297)|\tat org.apache.zookeeper.server.quorum.QuorumPeer.makeLeader(QuorumPeer.java:1272)|\tat org.apache.zookeeper.server.quorum.QuorumPeer.run(QuorumPeer.java:1479)", "metadata": {"container_name": "zookeeper", "pod_name": "no-metrics-pattern-zookeeper-0"}, "extra_data": {"file": "Leader.java", "line": "318", "mbkf_cluster_name": "no-metrics-pattern"}}

**Comment by vshinde727** [comment]
Did you find any pointers?
